src/OpenTelemetryEngine/Logging/FunctionsLogger.cs

```diff
@@ -0,0 +1,40 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.Extensions.Logging;
+using OpenTelemetryEngine.ResponseObjects;
+
+namespace OpenTelemetryEngine.Types
+{
+    public class FunctionsLogger
+    {
+        public ILogger logger;
+
+        public FunctionsLogger(ILogger logger) {  this.logger = logger; }
+
+        public void Log(object? logItem, string? level, Exception? exception = null) 
+        {
+            if (Enum.TryParse(typeof(LogLevel), level, out var _logLevel)) {
+                if (logItem is not null && _logLevel is not null) {
```

Comment:
Consider separating these two checks, so that you can throw a more specific exception that will unambiguously point to either the log message or the log level.

Also, it is more conventional to check for pre-conditions and throw exceptions first, and then proceed with the "happy case" logic. A code structured like this is easier to quickly scan. So, consider rewriting this as:

``` csharp
if (logItem is null)
{
    throw ...
}

if (_logLevel is null)
{
    throw ...
}

logger.Log(...)
```

Reply:
Done